fendermint/vm/upgrade_scheduler/src/upgrade_scheduler.rs

```diff
@@ -0,0 +1,91 @@
+// Copyright 2022-2024 Protocol Labs
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use std::collections::BTreeMap;
+
+use anyhow::Result;
+
+/// a function type for migration
+pub type MigrationFunc = fn(i64) -> i64;
+
+/// Upgrade represents a single upgrade to be executed at a given height
+#[derive(Clone)]
+pub struct Upgrade {
+    /// the block height at which the upgrade should be executed
+    pub height: i64,
+    /// the migration function to be executed
+    pub migration: MigrationFunc,
```

Comment:
Slightly curious that the upgrade knows when it's supposed to be applied, but the function takes the height as an argument too. It sounds like it should already know, as it's part of the upgrade itself?